internal/mq/msgstream/mq_factory.go

```diff
@@ -1,91 +1,24 @@
-// Licensed to the LF AI & Data foundation under one
-// or more contributor license agreements. See the NOTICE file
-// distributed with this work for additional information
-// regarding copyright ownership. The ASF licenses this file
-// to you under the Apache License, Version 2.0 (the
-// "License"); you may not use this file except in compliance
-// with the License. You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package msgstream
 
 import (
-	"context"
-
-	"go.uber.org/zap"
-
-	rmqimplserver "github.com/milvus-io/milvus/internal/mq/mqimpl/rocksmq/server"
-	rmqwrapper "github.com/milvus-io/milvus/internal/mq/msgstream/mqwrapper/rmq"
+	"github.com/milvus-io/milvus/internal/mq/mqimpl/rocksmq/server"
+	"github.com/milvus-io/milvus/internal/mq/msgstream/mqwrapper/rmq"
 	"github.com/milvus-io/milvus/pkg/log"
 	"github.com/milvus-io/milvus/pkg/mq/msgstream"
-	"github.com/milvus-io/milvus/pkg/mq/msgstream/mqwrapper"
+	"go.uber.org/zap"
 )
 
-// RmsFactory is a rocksmq msgstream factory that implemented Factory interface(msgstream.go)
-type RmsFactory struct {
-	dispatcherFactory msgstream.ProtoUDFactory
-	// the following members must be public, so that mapstructure.Decode() can access them
-	ReceiveBufSize int64
-	RmqBufSize     int64
-}
-
-// NewMsgStream is used to generate a new Msgstream object
-func (f *RmsFactory) NewMsgStream(ctx context.Context) (msgstream.MsgStream, error) {
-	rmqClient, err := rmqwrapper.NewClientWithDefaultOptions()
-	if err != nil {
-		return nil, err
+// NewRocksmqFactory creates a new message stream factory based on rocksmq.
+func NewRocksmqFactory(path string) msgstream.Factory {
+	if err := server.InitRocksMQ(path); err != nil {
+		log.Fatal("fail to init rocksmq", zap.Error(err))
```

Comment:
it's a common implementation of msgstream.Factory.
I move it to pkg/mq/msgstream/`CommonFactory`